src/annotator/sidebar.js

```diff
@@ -0,0 +1,326 @@
+      this.onLoginRequest = serviceConfig.onLoginRequest;
+      this.onLogoutRequest = serviceConfig.onLogoutRequest;
+      this.onSignupRequest = serviceConfig.onSignupRequest;
+      this.onProfileRequest = serviceConfig.onProfileRequest;
+      this.onHelpRequest = serviceConfig.onHelpRequest;
+    }
+
+    this.onLayoutChange = config.onLayoutChange;
+
+    // Initial layout notification
+    this._notifyOfLayoutChange(false);
+    this._setupSidebarEvents();
+  }
+
+  _setupSidebarEvents() {
+    annotationCounts(document.body, this.crossframe);
+    sidebarTrigger(document.body, () => this.show());
+    features.init(this.crossframe);
+
+    this.crossframe.on('showSidebar', () => this.show());
+    this.crossframe.on('hideSidebar', () => this.hide());
+
+    const eventHandlers = [
+      [events.LOGIN_REQUESTED, this.onLoginRequest],
+      [events.LOGOUT_REQUESTED, this.onLogoutRequest],
+      [events.SIGNUP_REQUESTED, this.onSignupRequest],
+      [events.PROFILE_REQUESTED, this.onProfileRequest],
+      [events.HELP_REQUESTED, this.onHelpRequest],
+    ];
+    eventHandlers.forEach(([event, handler]) => {
+      if (handler) {
+        this.crossframe.on(event, () => handler());
+      }
+    });
+  }
+
+  _resetGestureState() {
+    this._gestureState = { initial: null, final: null };
+  }
+
+  _setupGestures() {
+    const toggle = $(this.toolbar.sidebarToggleButton);
+
+    if (toggle[0]) {
+      // Prevent any default gestures on the handle.
+      toggle.on('touchmove', e => e.preventDefault());
+
+      // Set up Hammer instance and handlers.
+      const manager = new Hammer.Manager(toggle[0])
+        // eslint-disable-next-line no-restricted-properties
+        .on('panstart panend panleft panright', this._onPan.bind(this))
+        // eslint-disable-next-line no-restricted-properties
+        .on('swipeleft swiperight', this._onSwipe.bind(this));
+
+      // Set up gesture recognition.
+      const pan = manager.add(
+        new Hammer.Pan({ direction: Hammer.DIRECTION_HORIZONTAL })
+      );
+      const swipe = manager.add(
+        new Hammer.Swipe({ direction: Hammer.DIRECTION_HORIZONTAL })
+      );
+      swipe.recognizeWith(pan);
+    }
+  }
+
+  // Schedule any changes needed to update the sidebar layout.
+  _updateLayout() {
+    // Only schedule one frame at a time.
+    if (this.renderFrame) {
+      return;
+    }
+
+    // Schedule a frame.
+    this.renderFrame = requestAnimationFrame(() => {
+      this.renderFrame = null;
+
+      if (this._gestureState.final !== this._gestureState.initial) {
+        const margin = /** @type {number} */ (this._gestureState.final);
+        const width = -margin;
+        this.frame.css('margin-left', `${margin}px`);
+        if (width >= MIN_RESIZE) {
+          this.frame.css('width', `${width}px`);
+        }
+        this._notifyOfLayoutChange();
+      }
+    });
+  }
+
+  /**
+   * Notify integrator when sidebar is opened, closed or resized.
```

Comment:
I noticed this in my own forays into the coffeescript variant of this module: what does _integrator_ refer to here?

Reply:
Here it means the party that embedded the client in the current document. I'm not sure that we have a standardized term for that. Embedder perhaps?